File: aula 5/funcoes.cpp
#include <iostream>
#include <vector>
using namespace std;

string verificar_estoque_critico(int quantidade) {
    if (quantidade < 5) {
        return "[REPOSICAO NECESSARIA]";
    } else {
        return "";
    }
}

void exibir_cabecalho() {
    cout << "\n===== SORVETERIA DO DENER =====\n\n";
}

int main() {
    vector<string> nomes;
    vector<int> quantidades;

    for (int i = 0; i < 3; i++) {
        string nome;
        int qtd;

        cout << "Nome do produto: ";
        cin >> nome;

        cout << "Quantidade: ";
        cin >> qtd;

        nomes.push_back(nome);
        quantidades.push_back(qtd);
    }

    int num = 0;

    while (num != 2) {
        exibir_cabecalho();
        cout << "1 - Exibir relatorio\n2 - Sair\n";
        cin >> num;

        if (num == 1) {
            for (int i = 0; i < 3; i++) {
                string aviso = verificar_estoque_critico(quantidades[i]);
                cout << "Produto: " << nomes[i] << " " << aviso
                     << " | Estoque: " << quantidades[i] << endl;
            }
        }
    }

    return 0;
}

File: aula 5/funcoes.java
import java.util.ArrayList;
import java.util.Scanner;

public class funcoes {

    public static String verificar_estoque_critico(int quantidade) {
        if (quantidade < 5) {
            return "[REPOSICAO NECESSARIA]";
        } else {
            return "";
        }
    }

    public static void exibir_cabecalho() {
        System.out.println("\n===== SORVETERIA DO DENER =====\n");
    }

    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);

        ArrayList<String> nomes = new ArrayList<>();
        ArrayList<Integer> quantidades = new ArrayList<>();

        for (int i = 0; i < 3; i++) {
            System.out.print("Nome do produto: ");
            String nome = sc.next();

            System.out.print("Quantidade: ");
            int qtd = sc.nextInt();

            nomes.add(nome);
            quantidades.add(qtd);
        }

        int num = 0;

        while (num != 2) {
            exibir_cabecalho();
            System.out.println("1 - Exibir relatorio\n2 - Sair");
            num = sc.nextInt();

            if (num == 1) {
                for (int i = 0; i < 3; i++) {
                    String aviso = verificar_estoque_critico(quantidades.get(i));
                    System.out.println("Produto: " + nomes.get(i) + " " + aviso +
                            " | Estoque: " + quantidades.get(i));
                }
            }
        }
    }
}